File: src/model/request/order_book_response.rs
use derive_more::Constructor;
use serde::Deserialize;
use serde::Serialize;
use crate::model::AskBid;

/// This struct represents an order book response
#[derive(Serialize, Deserialize, Clone, Constructor, Debug, Default, PartialEq)]
pub struct OrderBookResponse {
    pub update_id: Option<u64>,
    pub last_update_id: Option<u64>,
    pub bids: Vec<AskBid>,
    pub asks: Vec<AskBid>,
}

File: bindings/csharp/OrderbookResponse.cs
namespace OpenLimits
{
    using System.Collections.Generic;

    public class OrderbookResponse
    {
        readonly public string market;
        readonly public IEnumerable<AskBid> asks;
        readonly public IEnumerable<AskBid> bids;

        readonly public ulong lastUpdateId;
        readonly public ulong updateId;

        public OrderbookResponse(string market, IEnumerable<AskBid> asks, IEnumerable<AskBid> bids, ulong lastUpdateId, ulong updateId)
        {
            this.market = market;
            this.asks = asks;
            this.bids = bids;
            this.lastUpdateId = lastUpdateId;
            this.updateId = updateId;
        }
    }
}